### ISSUE TITLE
Using this with lots of data on iOS Safari crashes website

### ISSUE THREAD

**Issue by schlagmichdoch** [gildas-lormeau/zip.js#391]
First of all: Thanks a lot for this awesome, lightweight library!

I implemented it similarily to the create file demo and got an issue that is reproducable on the demo as well:
https://gildas-lormeau.github.io/zip.js/demos/demo-create-file.html

Using iOS Safari:
1. Add more than 208 MB of files:
   Everything works fine: All Files are progressed successfully
2. Click "Download"
   Website crashes: A reload is forced after a blue banner on the top of the page appears for half a second that reads: "A problem occurred with this webpage so it was reloaded"

- When I add less than 200 MB everything works perfectly. 
- When it's more than 208 MB the page crashes when I try to download the Zip File.
- When I add ~1GB the page already crashes when progressing the files.

I guess it has to do with the maximum RAM a browser can occupy which is 384 MB on iOS Safari 15. Do you know of any workaround to that problem? Would it help to use the SplitDataReader class? I would appreciate any help!

**Comment by gildas-lormeau** [comment]
I don't have an iOS device and I was not able to reproduce the issue in the simulator bundled with Xcode.

I guess the Origin Private File System API could help, see https://webkit.org/blog/12257/the-file-system-access-api-with-origin-private-file-system/. However it's a bit tricky to use because [FileSystemFileHandle#createSyncAccessHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileHandle/createSyncAccessHandle) is only available in a Worker. The `SplitDataReader` would not be useful. However you could create split zip with by using a `SplitDataWriter` instance, it could help to save memory.

**Comment by schlagmichdoch** [comment]
Thanks for the quick response!

So your idea is to create a zip file on the device itself via the Private File System API and only upload it then? Not sure whether that would reduce memory usage though.

I decided to remove zip.js on iOS completely and go for another approach so the browser does not have to save the same blob twice. As I have the same problem when using navigator.share with big files though, I simply won't allow files bigger than 200 MB on iOS.

If you're interested where your project got implemented into: Peer to peer file sharing https://pairdrop.net/ ([Repo](https://github.com/schlagmichdoch/PairDrop))

As this issue is not really zip.js specific I'll close it, thanks for helping though!
